OPAL/br/src/main/scala/org/opalj/br/fpcf/analyses/FieldAccessInformationAnalysis.scala

```diff
@@ -0,0 +1,148 @@
+                    val fieldReadAccess = instruction.asInstanceOf[FieldReadAccess]
+                    resolveFieldReference(fieldReadAccess) match {
+                        case Some(field) =>
+                            readAccesses.getOrElseUpdate(field, new IntTrieSetBuilder()) += pc
+                        case None =>
+                            if (reportedFieldAccesses.add(instruction)) {
+                                val message = s"cannot resolve field read access: $instruction"
+                                OPALLogger.warn("project configuration", message)
+                            }
+                    }
+
+                case PUTFIELD.opcode | PUTSTATIC.opcode =>
+                    val fieldWriteAccess = instruction.asInstanceOf[FieldWriteAccess]
+                    resolveFieldReference(fieldWriteAccess) match {
+                        case Some(field) =>
+                            writeAccesses.getOrElseUpdate(field, new IntTrieSetBuilder()) += pc
+
+                        case None =>
+                            if (reportedFieldAccesses.add(instruction)) {
+                                val message = s"cannot resolve field write access: $instruction"
+                                OPALLogger.warn("project configuration", message)
+                            }
+                    }
+
+                case _ => /*nothing to do*/
+            }
+        }
+
+        val fieldAccesses = mutable.AnyRefMap.empty[Field, (List[(Method, PCs)], List[(Method, PCs)])]
+        readAccesses foreach { readAccess =>
+            val (field, pcs) = readAccess
+            fieldAccesses.get(field) match {
+                case Some(currentAccesses) =>
+                    fieldAccesses.put(field, ((method, pcs.result()) :: currentAccesses._1, currentAccesses._2))
+                case None =>
+                    fieldAccesses.put(field, ((method, pcs.result()) :: Nil, Nil))
+            }
+        }
+        writeAccesses foreach { writeAccess =>
+            val (field, pcs) = writeAccess
+            fieldAccesses.get(field) match {
+                case Some(currentAccesses) =>
+                    fieldAccesses.put(field, (currentAccesses._1, (method, pcs.result()) :: currentAccesses._2))
+                case None =>
+                    fieldAccesses.put(field, (Nil, (method, pcs.result()) :: Nil))
+            }
+        }
+
+        Results(fieldAccesses map { fieldAccess =>
+            val readAccesses = fieldAccess._2._1.to(Set)
+            val writeAccesses = fieldAccess._2._2.to(Set)
+
+            new PartialResult(
+                fieldAccess._1,
+                FieldAccessInformation.key,
+                (current: EOptionP[Field, FieldAccessInformation]) => current match {
+                    case InterimUBP(ub: FieldAccessInformation) =>
+                        if (readAccesses.subsetOf(ub.readAccesses) && writeAccesses.subsetOf(ub.writeAccesses))
```

Comment:
This might be more expensive than just performing the update anyway. One option might be to perform the set union, then check the resulting sets for identity (at least some set implementations return the original object if nothing was added)